src/Orleans.Transactions/State/TransactionQueue.cs

```diff
@@ -482,6 +482,10 @@ private async Task StorageWork()
                     {
                         // process all committable entries, adding storage events to the storage batch
                         CollectEventsForBatch(committableEntries);
+                        if (problemFlag != TransactionalStatus.Ok)
```

Comment:
`problemFlag` reminds me of [errno](http://man7.org/linux/man-pages/man3/errno.3.html) and [GetLastError](https://msdn.microsoft.com/en-us/library/windows/desktop/ms679360%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396), which makes me uncomfortable.

Reply:
Agree.  I openly acknowledge that this solution is kludgy, almost to the point of being a hack.  All three loops (lock worker, storage worker, and confirmation worker) need refactored.  I'm trying to fix things within the current code until I've time to do refactoring passes.